src/write/loc.rs

```diff
@@ -265,6 +277,23 @@ pub enum Location {
     },
 }
 
+fn write_expression<W: Writer>(
+    w: &mut W,
+    refs: &mut Vec<DebugInfoReference>,
+    encoding: Encoding,
+    unit_offsets: Option<&UnitOffsets>,
+    val: &Expression,
+) -> Result<()> {
+    let size = val.size(encoding, unit_offsets);
+    if encoding.version <= 4 {
+        w.write_u16(size as u16)?;
```

Comment:
```suggestion
        w.write_u16(u16::try_from(size)?)?;
```

To prevent silent truncation when it is bigger than `65_535` bytes.